### PR TITLE
libupnp: update to 1.14.14

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update + add C++11 standard (required).

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ctreleaven** [comment]
Somehow this got routed to my Junk mail so I just saw it.

Without trying, it appears you are not using the tarball from Releases and not using the bzip2 version?  Why not?

Where are you getting the cxx_standard?

I'm not enthused about the gratuitous changes to the portfile layout.  

Craig

**Comment by barracuda156** [comment]
> Without trying, it appears you are not using the tarball from Releases and not using the bzip2 version? Why not?

Something did not work, I need to re-check again.
By the way, version in releases and in tags are identical (from the same commit, [b25d2b5](https://github.com/pupnp/pupnp/commit/b25d2b556f19d33dbcf7589b0336f0c97d15c4da)): https://github.com/pupnp/pupnp/tags
https://github.com/pupnp/pupnp/releases

> Where are you getting the cxx_standard?

Here: https://github.com/pupnp/pupnp/blob/a3f40ec03898260433449c9d02ae6898cb4f449f/CMakeLists.txt#L6
Sure enough, configure errs out when `gcc-4.2` is picked by default due to not setting `cxx_standard`.

**Comment by barracuda156** [comment]
@ctreleaven So I tried using tarball from releases, and it is broken:
```
CMake Error: File /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_libupnp/libupnp/work/build/autoconfig.h.cm does not exist.
CMake Error at CMakeLists.txt:250 (configure_file):
  configure_file Problem configuring file


CMake Error at /opt/local/share/cmake-3.25/Modules/WriteBasicConfigVersionFile.cmake:43 (message):
  No VERSION specified for WRITE_BASIC_CONFIG_VERSION_FILE()
Call Stack (most recent call first):
  /opt/local/share/cmake-3.25/Modules/CMakePackageConfigHelpers.cmake:237 (write_basic_config_version_file)
  CMakeLists.txt:266 (write_basic_package_version_file)
```
The same version from tags configures and builds just fine.

File: devel/libupnp/Portfile [hunk 1]
```diff
@@ -4,28 +4,22 @@ PortSystem          1.0
 PortGroup           github 1.0
 PortGroup           cmake 1.1
 
-github.setup        pupnp pupnp 1.14.12 release-
-
-github.tarball_from releases
```
Comment (reneeotten) [inline devel/libupnp/Portfile:9]: please keep using "releases"; you'll also need to put "use_bzip yess" back. And do not make unnecessary whitespace changes for maintained ports. Minimal changes please to do the update.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: devel/libupnp/Portfile (hunk 1 above)
Comment (barracuda156) [reply devel/libupnp/Portfile:9]: The releases tarball is broken, I have pointed out at that in comments.

Comment (reneeotten) [reply devel/libupnp/Portfile:9]: I'll take a look as that sounds strange... in the meantime please revert unneeded formatting changes (i.e., whitespace or reordering of lines)

**Comment by barracuda156** [comment]
@reneeotten I have removed formatting changes.

P. S. Bzip does not work from tags: `bzip2: /opt/local/var/macports/distfiles/libupnp/pupnp-1.14.14.tar.bz2 is not a bzip2 file`. And releases version fails to configure: https://github.com/macports/macports-ports/pull/16462#issuecomment-1301557518

**Comment by ctreleaven** [comment]
Issue raised upstream re missing files in the tarball:

https://github.com/pupnp/pupnp/issues/419

**Comment by barracuda156** [comment]
> Issue raised upstream re missing files in the tarball:
> 
> [pupnp/pupnp#419](https://github.com/pupnp/pupnp/issues/419)

@ctreleaven Should I close this PR?

**Comment by ctreleaven** [comment]
Let's give it a couple of days and see if they'll do a new release correcting the problem.  Presumably other builders will also be affected.

**Comment by ctreleaven** [comment]
Upstream issued a new release adding the missing cmake files.  Seems to be working now.
